app/src/main/java/org/mozilla/focus/session/SessionCallbackProxy.java

```diff
@@ -56,6 +56,10 @@ public void onProgress(int progress) {
     @Override
     public void onURLChanged(String url) {
         session.setUrl(url);
+        if (session.isSearch()) {
+            session.setSearch(false);
+            session.setSearchUrl(url);
+        }
```

Comment:
Wouldn't this break in cases where the search page redirects to a different URL (e.g. Wikipedia)?

Reply:
nit: I think I saw this snippet a couple of times in our code base now. Maybe it's time for a session.clearSearchIfNeeded() method or something like that. :)

Reply:
Ugh, yes it would probably break for redirects. Any ideas? Should we scrap the URL matching altogether?

Reply:
Oh, do we still do URL matching? I somehow missed that :D

Reply:
Not regex matching, just literally comparing two URLs. That was the solution I implemented so that we could keep track and compare the search URL with current URL to see if it was a search (after clicking away from the URL bar). But I guess I forgot to account for if the user clicks away without clicking the URL bar...  >.<
https://github.com/mozilla-mobile/focus-android/blob/master/app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.java#L187

Reply:
Ah, the WebView might be able to tell us whether a new URL came from a user interaction. I guess we could just keep the search terms (and display them) until there's a new request triggered by a user interaction (like a click on a link) - and completely ignore the URL? I'll write a patch and you can review it if you have time for it. :)